docs/v2_design.md

```diff
@@ -2,188 +2,600 @@
 
-static_assert(isq::stress_dim / isq::stress_dim != isq::strain_dim);                // ???
-static_assert(isq::stress_dim / isq::stress_dim != isq::efficiency_dim);            // ???
-static_assert(equivalent(isq::stress_dim / isq::stress_dim, isq::strain_dim));
-static_assert(equivalent(isq::stress_dim / isq::stress_dim, isq::efficiency_dim));  // ???
-static_assert(convertible(isq::stress_dim / isq::stress_dim, isq::strain_dim));
-static_assert(convertible(isq::stress_dim / isq::stress_dim, isq::efficiency_dim)); // ???
-```
 
----
+## Quantity Characteristics
 
-## Units
+ISO 80000 defines not only scalar quantities. Some of them are defined to have vector or tensor
+characteristics. To model that the library defines:
 
 ```cpp
-// Different magnitudes
-static_assert(si::metre != si::kilometre);
-static_assert(!equivalent(si::metre, si::kilometre));
-static_assert(convertible(si::metre, si::kilometre));
-
-// Unnamed derived unit vs named unit
-static_assert(si::newton * si::metre != si::joule);
-static_assert(equivalent(si::newton * si::metre, si::joule));
-static_assert(convertible(si::newton * si::metre, si::joule));
+enum class quantity_character { scalar, vector, tensor };
 ```
 
----
+The above assumes the following hierarchy:
+- every quantity can be expressed with a scalar representation type
+- vector representation type can be provided only to quantities with vector or tensor characteristics
+- tensor representation type can be used only for tensor quantities
 
-## Conversions
+The support for the above is still WIP. However, we assume that there will be variable template
+customization points for representation types that a user will have to specialize for a custom types.
 
-```cpp
-// Implicit conversions allowed between quantities of `convertible` references
-auto q1 = 120 * si::length[km] / (2 * si::time[h]);
-quantity<si::speed[km / h]> s = q1;
+The default values for type traits will be as follows:
 
-// Explicit casts allow changing all or only a part of the type
-auto q2 = quantity_cast<isq::speed_dim>(120 * si::length[km] / (2 * si::time[h]));
-auto q3 = quantity_cast<m / s>(120 * si::length[km] / (2 * si::time[h]));
-auto q4 = quantity_cast<si::speed[m / s]>(120 * si::length[km] / (2 * si::time[h]));
-auto q5 = quantity_cast<double>(120 * si::length[km] / (2 * si::time[h]));
-auto q6 = quantity_cast<quantity<si::speed[m / s], double>>(120 * si::length[km] / (2 * si::time[h]));
+```cpp
+template<typename T>
+inline constexpr bool is_scalar = false;
+
+template<typename T>
+  requires std::is_floating_point_v<T> ||
+             (std::is_integral_v<T> && one_of_types<T,
+                                                    signed char, short, int, long, long long,
+                                                    unsigned char, unsigned short, unsigned,
+                                                    unsigned long, unsigned long long>())
```

Comment:
Why the explicit list? To exclude `bool` and `char`? `std::int8_t` and `std::uint8_t` are not required to not be `char`. This also excludes extended types.

Reply:
Yes, I think that quantities of `bool` and `char` should not be allowed by default. But you made a good point about `cstdint` types. Should I just explicitly exclude `bool` then?

Reply:
Yeah, I think that's consistent with what the C++ standard library does.

Reply:
https://wg21.link/LWG3838 will add just the right thing.